### PR TITLE
qcalc spawning behavior change

### PR DESCRIPTION
**Pull request by bootstrap-prime**
I changed the behavior of calc_preprocess_input to only call qcalc when the input string changes. This is a partial solution to #40. I'm still learning C so I may or may not have a better solution for this later on.

### CODE REVIEW

File: src/calc.c [hunk 1]
```diff
@@ -522,38 +524,44 @@ static char* calc_preprocess_input(Mode* sw, const char* input)
...
+        }
 
-    g_subprocess_wait_check_async(process, NULL, process_cb, (gpointer)&pd->last_result);
+        g_subprocess_wait_check_async(process, NULL, process_cb, (gpointer)&pd->last_result);
 
-    return g_strdup(input);
+        return g_strdup(input);
+    } else {
+        return g_strdup(pd->previous_input);
+    }
```
Comment (svenstaro) [inline src/calc.c:561]: I think this whole thing could be done more elegantly using an early return. What do you think?

**Comment by svenstaro** [comment]
@bootstrap-prime Are you interested in following up on my little review?

**Comment by bootstrap-prime** [comment]
absolutely, sorry for the delay.

File: src/calc.c (hunk 1 above)
Comment (bootstrap-prime) [reply src/calc.c]: sure, that sounds good. I have an idea on how to improve this further though, if I can find a way to make qalc not fork gnuplot or dynamically update it.

**Comment by bootstrap-prime** [comment]
I sort of figured out how to make this better - spawn an interactive qcalc session and then pipe up inputs and outputs.
Unfortunately, it seems that gio g_input_stream_* doesn't have the ability to read, and then return after a timeout, and I'm not familiar enough with async here to implement a cancel.
I thought I could get around this by just directly using libqalculate, but it's C++ and this project is pure C. libqalculate may expose a C abi, but I'm not sure how to use it or what to look for.

**Comment by svenstaro** [comment]
Shall I take a proper look at this PR or do you wanna research this topic further first?

**Comment by bootstrap-prime** [comment]
I don't think I have the motivation to continue, honestly. Most of the window opening spam is removed, so this is probably a good stopping point.

**Comment by svenstaro** [comment]
Sorry for forgetting about this for so long. Finally merged it. Thanks!

**Pull request merged**